### ISSUE TITLE
Output N scores, one per node

### ISSUE THREAD

**Issue by Richard549** [tkipf/gcn#84]
Hi, I'm a little confused and hoping you could offer some insight.

In the implementation, it seems like each model input is a single node, and each model output is a vector of label probabilities for that particular node.

Is a GCN able to be used for a task in which I want to generate a probability distribution across the graph nodes? That is, if my graph has N nodes each with M features, then the input layer consists of (N*M) neurons, and the output (softmax) layer consists of N neurons?

Each node would have one output feature, but all the output feature values would sum to 1. I'm finding it hard to reconcile this idea with the implementation, but perhaps I'm misunderstanding something.

Any thoughts would be much appreciated, thanks!

**Comment by tkipf** [comment]
Yes, such a setting is no problem, just set the output dimensionality to 1
and apply a softmax across nodes (instead of across class logits of a
single node).

**Comment by Richard549** [comment]
Thanks for getting back to me. In order to softmax across nodes, then I should let my GCN output (input_dim = M, output_dim = 1) be the 'raw' logit value, right? I then collect these raw outputs across nodes, and softmax to get probabilities.

I think this makes sense, thanks very much!

Final quick-question: what is the meaning of 'support' and 'num_supports'?
